### ISSUE TITLE
Adds support for transform.matrix on DisplayObject, also fixes some weird matrix issues

### ISSUE THREAD

**Issue by jgranick** [openfl/openfl-html5#12]
#12 Issue by __fserb__,

For some reason the last pull request disappeared when merged.
Here it is again in a new branch and already merged with the latest changes.

**Comment by jgranick** [comment]
Comment by __jgranick__:

Thank you!

I've taken your improvements, and made some changes, and just committed them, I decided not to hold onto a Transform instance unless this is accessed off the DisplayObject, we could decide not to hold onto it even then (and instead to just return an instance with the getter) but maybe this is best to reduce garbage collection.

I updated the transform behavior, I noticed that you can save an object's transform and poll it periodically to get the current transform for that object. As a result, I decided to make matrix a getter and a setter, to pull the current matrix off the display object (but not using worldTransform, since that may not be current when calling this, and we don't want the world matrix) or to set the DisplayObject's values.

The current implementation in working in my tests, but if you would not mind taking a look, I'd really appreciate your feedback and confirmation if this works for you. Thanks!

**Comment by jgranick** [comment]
Comment by __fserb__:

That makes a lot of sense, is cleaner and works better. As I said in the original request, I had no idea what I was doing, it was a typical fix hack. The reason I did that way was following the way openfl-bitfive handles it, but this makes much more sense. I had noticed that there was something weird with worldTransform but was going to take a look later.

 I've tested with my games and they all seem to work fine.

It seems it ends up doing both computations every time the matrix is set: i.e., we convert from matrix to scale/rotation/trans of the DisplayObject and then back to the worldTransform scale/rotation/trans. Would it make sense for the DisplayObject to keep and update its transform object on the setters of scale/rotation/trans and use the matrix instead of the scale/rotation/trans values? Maybe it's too error prone.

Anyways, thanks for fixing it for real! My games are almost in par with the flash/native versions. Missing sound (will get to that in a bit) and there is something weird with the quality of Graphics.
